### DISCUSSION TITLE
Redesigned version

### DISCUSSION THREAD

**Discussion by wtfruchit** [subhra74/xdm#528] [Show and tell]
_Redesigned UX/UI of the software, making it more user friendly and aesthetically appealing!_
### Dashboard
![Dashboard](https://user-images.githubusercontent.com/42517652/114157325-281f5c80-9941-11eb-9caf-abe6c695ad02.png)
### Dashboard-Selected
![Dashboard-Selected](https://user-images.githubusercontent.com/42517652/114157330-29508980-9941-11eb-9b1f-a8ebd6feecb6.png)
### Toolbar options
![Toolbar options](https://user-images.githubusercontent.com/42517652/114157336-29e92000-9941-11eb-9ae2-105610c4a6a7.png)
### Downloading Screens
**Downloading in progress**
![Downloading-state](https://user-images.githubusercontent.com/42517652/114157636-792f5080-9941-11eb-8e1d-c27854144109.png)
**Downloading Paused**
![Paused-state](https://user-images.githubusercontent.com/42517652/114157638-792f5080-9941-11eb-88da-ca7d748d14bb.png)
**Downloading complete**
![Download complete - state](https://user-images.githubusercontent.com/42517652/114157634-77fe2380-9941-11eb-9739-591f1893a2b1.png)

**Comment by ARotterGoodMan** [comment]
I feel OK

**Comment by unexploredtest** [comment]
Looks really nice! Have you released it?

**Comment by subhra74** [comment]
@Ruchit4601 , The redesigned UI looks really great, color combination and aesthetics are awesome. Unfortunately we have several constraints that would prevent us from using an UI like above. 
Main constraints would be:

- We would like to keep the apps binary size and memory usage footprint to minimal so I am in a process of moving XDM to dotnet core and have a UI built in WinForm and Gtk for Windows and Linux/Mac. With WinForm it is difficult use use advanced effects like drop shadow or transparent gradient ( I personally wanted to move to flutter but that would require too much time and effort )
- While re-designing we would like to make the UI more compact so that users with low screen size/resolution can also have a decent user experience. Something that I started working on but did not progress much due to lack of time.

This is something I started working on, and it would be really great if we can apply some of the styles/color schemes in it for the main window. The download progress window that you designed looks perfect and would not need any change. The download list that you designed looks great but will require lot of effort to implement in WinForm and Gtk, so better would be to use a table/grid instead.

This would the new compact UI, but currently it lacks icons and a nice color scheme
![image](https://user-images.githubusercontent.com/8403090/114263266-cfa19b00-99e4-11eb-9ca9-12febc8cc5e4.png)

**Comment by moedwyn** [comment]
Whoa,,, its beautifull i like it,
but if the application of this design makes xdm heavy, it's better I give advice so that the design is made simple but light hehehe....
overall it's good.... 👍👍👍

**Comment by muhammadfaruq72** [comment]
![Screenshot 2021-06-17 at 2 03 18 PM](https://user-images.githubusercontent.com/51035641/122366374-f0521800-cf74-11eb-9f01-384d968561f0.png)

This application is fully working on macOS Big Sur including XDM Safari extension. Only waiting for a brand new UI of XDM.

**Comment by adamlahbib** [comment]
Very Nice UI, it should have a dark mode as well

**Reply by IPings** [reply]
Yes, please! A dark mode is a must.

**Comment by JihanLoong** [comment]
I love this UI, it puts all the information needed in order on deck, so it is easy to read.

**Comment by Borwe** [comment]
Pretty, but what a big window it must be.

**Comment by unexploredtest** [comment]
Maybe we should create a fork? The owner seems busy.

**Comment by subhra74** [comment]
As of now it does not look very feasible to implement the theming which was suggested at the beginning of this discussion, currently a major rewrite of XDM is underway and I am hoping to merge the new to to xdm main repo soon and be able to release new beta version by end of October.

**Reply by unexploredtest** [reply]
Thanks for the heads up! Waiting for the release.

**Comment by subhra74** [comment]
New UI will mostly look like this:
Light Theme
![Capture-1](https://user-images.githubusercontent.com/8403090/138966235-a3610da8-5113-4e81-912f-5079cab75949.PNG)
![Capture-2](https://user-images.githubusercontent.com/8403090/138966271-6fa70b7d-0aa7-4fa0-bc88-b30eb07da352.PNG)
Dark Theme
![Capture-3](https://user-images.githubusercontent.com/8403090/139157364-a0af6a4a-6089-44cb-a730-68badf2dff9c.PNG)
![Capture-4](https://user-images.githubusercontent.com/8403090/139157386-49a19338-4c96-4a5e-8190-c08393c9bfba.PNG)

**Reply by unexploredtest** [reply]
Nice! So version 8 it is. Does it also contain Dark Mode?

**Reply by subhra74** [reply]
Updated with dark mode

**Reply by unexploredtest** [reply]
Looks neat

**Reply by adamlahbib** [reply]
yeah it looks neat! while still in build can you add Speed Limiter for current download option and make the file draggable once download is complete like IDM @subhra74 
![image](https://user-images.githubusercontent.com/37028131/139556280-076071d2-cbc8-45f8-a5d3-39f5774d38f0.png)
![image](https://user-images.githubusercontent.com/37028131/139556287-2010b3bb-cc7e-4889-b96b-1c5e8caf09b7.png)

**Reply by ARotterGoodMan** [reply]
very good
